admin_ui/src/components/CSVButton.vue

```diff
@@ -59,21 +55,25 @@ export default {
                     )
                     exportedRows.push(...response.data.rows)
                 }
-                return exportedRows
+                let csv: string = "data:text/csv;charset=utf-8,"
+                csv += [
+                    Object.keys(exportedRows[0]).join(";"),
+                    ...exportedRows.map((item) => Object.values(item).join(";"))
+                ]
+                    .join("\n")
+                    .replace(/(^\[)|(\]$)/gm, "")
+                const data: string = encodeURI(csv)
+                const link: HTMLAnchorElement = document.createElement("a")
+                link.setAttribute("href", data)
+                link.setAttribute("download", `${this.tableName}.csv`)
+                link.click()
```

Comment:
OK, I think I understand what's happening here.

Does this created link appear anywhere in the document? Should it be deleted afterwards. Or just have a permanent hidden link, which we set the download and href attributes on using Vue.

Reply:
Yes, this creates a link with data and download attributes and the behavior is the same as in `vue-json-excel`. Feel free to change this as you think is best or we can always go back to using an external library.